docs/4.1/architecture/nodes.md

```diff
@@ -0,0 +1,109 @@
+## Teleport Nodes
+
+[TOC]
+
+## The Node Service
+
+A regular node becomes a Teleport Node when the node joins a cluster with an
+"join" token. Read about how nodes are issued certificates in the
+[Auth Guide](./auth/#issuing-node-certificates).
+
+![Node joins a cluster](../img/node_join.svg)
+
+A Teleport Node runs the [`teleport`](../cli-docs/#teleport) daemon with the
+`node` role. This process handles incoming connection requests, authentication,
+and remote command execution on the node, similar to the function of OpenSSH's
+`sshd`.
+
+![Node Service ping API](../img/node_service_api.svg)
+
+All cluster Nodes keep the Auth Server updated with their status with periodic
+ping messages. They report their IP addresses and values of their assigned
+labels. Nodes can access the list of all Nodes in their cluster via the
+[Auth Server API](./auth/#auth-api).
+
+!!! tip "Tip"
+    In lightweight environments, such as IoT deployments, it is possible to entirely replace `sshd` with the `node` service <!--other examples?-->.
```

Comment:
I think we would advise always replacing `sshd` with the Teleport node service unless there's a really compelling reason not to (like existing workflows relying on `ssh`, embedded devices that can't run a custom binary, etc)